CONTRIBUTING.md

```diff
@@ -0,0 +1,334 @@
+# Contributing to caugi
+
+Thank you for your interest in contributing to `caugi`! This document provides guidelines and information to help you contribute effectively to the project.
+
+## Table of Contents
+
+- [Project Scope and Overview](#project-scope-and-overview)
+- [Getting Started](#getting-started)
+- [Development Environment Setup](#development-environment-setup)
+- [Architecture](#architecture)
+- [Code Style Guidelines](#code-style-guidelines)
+- [Testing](#testing)
+- [Submitting Changes](#submitting-changes)
+- [Reporting Issues](#reporting-issues)
+
+## Project Scope and Overview
+
+`caugi` (pronounced "corgi") is a **Causal Graph Interface** package for R, providing a high-performance, tidy toolbox for building, coercing, and analyzing causal graphs. The package is designed to be:
+
+- **Causality-first**: Focused on causal graph operations and algorithms
+- **High-performance**: Leveraging Rust for performance-critical operations
+- **Tidy**: Following tidyverse principles and conventions
+- **Flexible**: Supporting multiple graph types and custom edge definitions
+
+### Supported Graph Classes
```

Comment:
Don't add these specifics about what is currently supported. We shouldn't add maintenance burden of keeping this up to date.